### DISCUSSION TITLE
first attempt ... checking

### DISCUSSION THREAD

**Discussion by pterusgithub** [WannesMalfait/Blender-Add-ons#26] [Q&A] [answered]
First of all, I'm new to using blender, geometry nodes and python...
The following example works but I would be very interested if you could tell me if there is a better way to achieve the same result. Just as a check ... Many thanks 
```
lenght = 20;
point_number = 100;
x,y = position();
curve = curve_line(end={lenght,0,0}).resample_curve(count=point_number).set_position(offset={0,sin(x),0});
profil = curve_circle(radius=0.2);
curve_to_mesh(curve,profil);
```

**Comment by WannesMalfait** [comment, marked as answer]
That's how I would do it as well. 

BTW: If you are interested in really creating whole setups with "code", this add-on might be more suitable for you:

https://github.com/carson-katri/geometry-script

**Reply by pterusgithub** [reply]
P.S. : my first main objective is to create (by using geometry nodes) a 3D model of a liam f1 (type Archimedes spiral, see image)  which would allow me to play with different parameters: 
- height 
- width 
- thickness 
- angles 
- ...
 
I guess that your add-on will help me ...

![image](https://github.com/WannesMalfait/Blender-Add-ons/assets/103185786/1e0ea22f-ed3e-4328-913b-6a06b0d661ec)

**Reply by WannesMalfait** [reply]
Interesting. This looks like an ideal use case for geometry nodes. Good luck!
